source/extensions/dynamic_modules/abi.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+// NOLINT(namespace-envoy)
+
+// This is a pure single C header file that defines the ABI of the dynamic module
+// used by Envoy.
+//
+// This must not contain any dependencies besides standard library since it is not only used by
+// Envoy itself but also by dynamic module SDKs written in non-C++ languages.
+//
+// Currently, compatibility is only guaranteed by an exact version match between the Envoy
+// codebase and the dynamic module SDKs. In the future, after the ABI is stabilized, we will revisit
+// this restriction and hopefully provide a wider compatibility guarantee. Until then, we use the
+// simple symbol prefix "envoy_dynamic_module_" without any versioning at the symbol level.
+//
+// All ABI definitions, regardless of the area of usage, must be included in this single header
+// file. This is not only to ensure easy inclusion by any language bindings but also because the ABI
```

Comment:
I'm not sure I understand how some of the requirements below lead to requiring a single header file. Is this really required? I can see this become unwieldy over time.

Reply:
sorry, you are correct in that this is not a requirement indeed, and I share the concern about this getting huge in the future. This is mostly about making the code simple (`#include "abi.h"` and done rather than multiple includes and build settings), but now I am think it's better to have separate abi headers. let me fix the comment.